### ISSUE TITLE
Issues with Training our humanoid to Walk

### ISSUE THREAD

**Issue by KarthiU** [UW-RoboSoccer/soccer_env#3]
**Working in the `walking_ernest` branch** 

We have currently been training models using `walker_ernest.ipynb`, and have so far been trying to replicate the results seen in [mujoco's humanoid training](https://colab.research.google.com/github/google-deepmind/mujoco/blob/main/mjx/tutorial.ipynb#scrollTo=ObF1UXrkb0Nd) but repurposed for our own xml. 

However, we have been unable to get the robot to walk with a natural gait, and were wondering if the issues lies in how the xml is structured (`ernest_humanoid_mod.xml`) as some of the physics seem off, or if the problem is more due to our reward and hyperparameter tuning. 

For reference our reward system is as follows (found in `envs/walking_rewards.py`) 
1. Standing Reward: +  for keeping the Torso of the robot within a window (0.4-0.6m, torso height defaults at 0.5) 
2. Upright reward: + for keeping the Torso of the robot upright (previously ran into issues where the robot would flip over and use its arms to maintain the height of the Torso)
3. Forward reward: + for moving the robot in the forward x direction 
4. Control cost: - penalizes large changes 

Note that there is also a rotation reward, but it currently is weighted as 0.

Here's how the current robot moves and the associated reward metrics graphs  

https://github.com/user-attachments/assets/ff520cb5-b4d1-4d18-99d8-feb490a5a937

<img width="419" alt="Image" src="https://github.com/user-attachments/assets/78648882-4cea-4381-98cb-56f5adeaa87c" />

**Comment by willthibault** [comment]
@KarthiU, can you please indicate the commit/branch you are referring to? I have no idea what files you are actually referring to.

Looking at the rewards, I suspect that the weightings are a bit strange. I would start with weights similar to the example then tune from there. I also would suggest looking at other reward schemes like those in the papers I have added to the RL locomotion folder in the Zotero library or MuJoCo playground because the one you are following is for a very primitive MuJoCo and is unrealistic for a real robot.

One simplification that you should start with is limiting your action space to just be the leg joints. Pick a fixed walking posture then just learn the leg joint motions. You will need to define the size of your action space with something like this:
```
  # Change the action size to be only the number of controlled joints for the RL policy
  @property
  def action_size(self) -> int:
    return self._controlled_joints
```

Also, the reward tracking is a good start, but you should switch to the W&B setup I suggested. This way you can systematically change the rewards and see how it behaved compared to the last set of rewards (change one at a time to get a sense of what helps and what doesn't). Reward tuning can be a bit of an art so this may take a lot of time.

**Comment by KarthiU** [comment]
Ok sounds good, I'll look into implementing the changes as you mentioned. Once I fix up the reward system based on the Zotero papers added I'll setup the repo on W&B. 

Also, I mentioned it in the previous issue, but just to reiterate, we are **working in the `walking_ernest` branch**.
